llvm/lib/IR/AsmWriter.cpp

```diff
@@ -1683,8 +1684,28 @@ static const char *getLinkagePrintName(GlobalValue::LinkageTypes LT) {
 
 void MDFieldPrinter::printLinkage(GlobalValue::LinkageTypes LT) {
   Out << FS << "linkage: ";
-  StringRef Linkage = StringRef(getLinkagePrintName(LT)).rtrim(' ');
-  Out << (Linkage.empty() ? "external" : Linkage);
+  const std::string Linkage = getLinkagePrintName(LT);
+  size_t end = Linkage.find_last_not_of(" ");
+  Out << (end == std::string::npos ? "external" : Linkage.substr(0, end + 1));
```

Comment:
If `find_last_not_of()` fails, `end` will be `npos` and adding 1 to `npos` probably isn't right.

Reply:
If `end` is equal to `npos`, a string of `external` is returned and `end+1` is not called.

Reply:
Oh yes, so it is.